Client/Frontend/Home/BookmarksEditPanel.swift

```diff
@@ -0,0 +1,335 @@
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        self.tableView.accessibilityIdentifier = "Bookmarks List"
+        self.tableView.backgroundColor = UIColor(red: 240/255, green: 240/255, blue: 240/255, alpha: 1)
+    }
+    
+    func deletePressed() {
+        guard let source = self.source,
+              let bookmark = self.bookmark else {
+                return
+        }
+        
+        guard let factory = source.modelFactory.value.successValue else {
+            log.error("Couldn't get model factory. This is unexpected.")
+            self.onModelFailure(DatabaseError(description: "Unable to get factory."))
+            return
+        }
+        
+        if let err = factory.removeByGUID(bookmark.guid).value.failureValue {
+            log.debug("Failed to remove \(bookmark.guid).")
+            self.onModelFailure(err)
+            return
+        }
+        
+        NSNotificationCenter.defaultCenter().postNotificationName(BookmarkStatusChangedNotification, object: bookmark, userInfo:["added": false])
+        self.navigationController?.popViewControllerAnimated(true)
+    }
+    
+    private func onModelFailure(e: Any) {
+        log.error("Error: failed to get data: \(e)")
+    }
+    
+    override func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        switch section {
+        case 0:
+            return DetailPanel._numberOfPanels.rawValue
+        case 1:
+            return 1
+        default:
+            return 0
+        }
+    }
+    
+    override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+        switch indexPath.section {
+        case 0:
+            let cell = tableView.dequeueReusableCellWithIdentifier(BookmarksEditPanelCellIdentifier, forIndexPath: indexPath) as! BookmarksEditPanelCell
+            switch DetailPanel(rawValue: indexPath.row) ?? .Title {
+            case .Title:
+                cell.label.text = NSLocalizedString("Title", comment: "The title of a bookmark.")
+                cell.textField.text = bookmark?.title
+                if titleField == nil {
+                    titleField = cell.textField
+                    cell.textField.addTarget(self, action: #selector(BookmarksEditPanel.textFieldDidChange(_:)), forControlEvents: UIControlEvents.EditingChanged)
+                    cell.textField.delegate = self
+                }
+            case .URL:
+                cell.label.text = NSLocalizedString("URL", comment: "URL of a bookmark.")
+                cell.textField.text = bookmark?.url
+                if urlField == nil {
+                    urlField = cell.textField
+                    cell.textField.addTarget(self, action: #selector(BookmarksEditPanel.textFieldDidChange(_:)), forControlEvents: UIControlEvents.EditingChanged)
```

Comment:
Since `textFieldDidChange` is the delegate method in `UITextFieldDelegate` and we're assigning ourselves as the delegate we don't need to explicitly call `addTarget` since the delegate will get called from `UITextField`.